x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/get_prepackaged_rules_status_route.ts

```diff
@@ -50,12 +52,23 @@ export const createGetPrepackagedRulesStatusRoute = (
         const prepackagedRules = await getExistingPrepackagedRules({ alertsClient });
         const rulesToInstall = getRulesToInstall(rulesFromFileSystem, prepackagedRules);
         const rulesToUpdate = getRulesToUpdate(rulesFromFileSystem, prepackagedRules);
-        return {
+        const prepackagedRulesStatus: PrePackagedRulesStatusSchema = {
           rules_custom_installed: customRules.total,
           rules_installed: prepackagedRules.length,
           rules_not_installed: rulesToInstall.length,
           rules_not_updated: rulesToUpdate.length,
         };
+        const validate = validateAddPrePackagedRulesStatus(prepackagedRulesStatus);
```

Comment:
My preferred API would be:

```ts
const [errors, value] = validate(prepackagedRulesStatus, prepackagedRulesStatusSchema)
```

rather than having to define a new function for each schema, if that's possible. Maybe we'd want the schema to be the first argument for currying?

Reply:
I might be able to do that. Let me try that out with this PR and get back to you if I can't. I should be able to inject the schema and use that which would down on me having to write unit tests and boiler plate code.

Reply:
Ok, I swapped the arrays to be value first, error second but I think I got everything we would want. 

I was able to clean up a lot of code with this suggestion and I am also returning the correct type and I no longer return the "transformed" as-is so I think this gets us really close to the original decode feature of io-ts. 

Appreciate the feedback 👍 

<img width="742" alt="Screen Shot 2020-02-24 at 7 32 11 PM" src="https://user-images.githubusercontent.com/1151048/75214475-f94d2380-574a-11ea-9e8d-465f8e297e9b.png">

Reply:
@FrankHassanabad  what's the value added with swapping the array to be value first, error second? I kind of liked it the other way as it follows the left=errors, right=response as well as error first aspect of nodejs. Just curious!